sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,233 @@
+      });
+  });
+
+  it(`Should get application details with scholastic standing change type when application has a scholastic standing '${StudentScholasticStandingChangeType.StudentDidNotCompleteProgram} associated with.`, async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(appDataSource, {
+      student,
+    });
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+    const [firstDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    // Create a scholastic standing and have it associated with the completed application.
+    const scholasticStanding = createFakeStudentScholasticStanding({
+      submittedBy: submittedByInstitutionUser,
+    });
+    // 'Student did not complete program' is the only'scholastic standing that does not generate an assessment.
+    // The below record has only a relationship with the application which must be enough to
+    // have the scholasticStandingChangeType returned.
+    scholasticStanding.application = application;
+    await studentScholasticStandingRepo.save(scholasticStanding);
+
+    const endpoint = `/students/application/${application.id}/completed`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        firstDisbursement: {
+          coeStatus: firstDisbursement.coeStatus,
+          disbursementScheduleStatus:
+            firstDisbursement.disbursementScheduleStatus,
+        },
+        assessmentTriggerType: application.currentAssessment.triggerType,
+        scholasticStandingChangeType:
+          StudentScholasticStandingChangeType.StudentDidNotCompleteProgram,
+      });
+  });
+
+  it(`Should get application details with the most updated appeal status when the application has more than one student appeals associated with.`, async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(appDataSource, {
+      student,
+    });
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+    const [firstDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    // Create approved student appeal
+    const approvedAppealRequest = createFakeStudentAppealRequest();
+    approvedAppealRequest.appealStatus = StudentAppealStatus.Approved;
+    const approvedAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [approvedAppealRequest],
+    });
+    approvedAppeal.submittedDate = addDays(-1);
```

Comment:
mm, here `approved appeal` is submitted yesterday, and from the at line `201` declined appeal is also submitted yesterday, but I didn't see the logic for that in declined appeal. let me know if iam missing anything

Reply:
I changed it and did not update the comment. I will change the comment.

Reply:
Comment updated.